src/openrct2/world/MapGen.cpp

```diff
@@ -279,9 +279,9 @@ static bool MapGenSurfaceTakesSnowTrees(const TerrainSurfaceObject& surface)
  */
 static void mapgen_place_trees()
 {
-    std::vector<int32_t> grassTreeIds(std::size(GrassTrees), 0);
-    std::vector<int32_t> desertTreeIds(std::size(DesertTrees), 0);
-    std::vector<int32_t> snowTreeIds(std::size(SnowTrees), 0);
+    std::vector<int32_t> grassTreeIds(std::size(GrassTrees), OBJECT_ENTRY_INDEX_NULL);
```

Comment:
what is this code doing? I can't understand why it is prepopulating it with nulls.

Reply:
Yup just checked this and this is wrong. Can you delete the initialisation of this with any value. The vectors should be all 0. Its artificially reducing the amount of trees.

Reply:
As I had everything running I've made the changes myself and added it as an additional commit.